backend/howtheyvote/worker/worker.py

```diff
@@ -175,13 +175,31 @@ def schedule_pipeline(
         hours: Iterable[int] = {0},
         minutes: Iterable[int] = {0},
         tz: str | None = None,
+        idempotency_key: Callable[..., str] | None = None,
     ) -> None:
         """Same as `schedule`, but handles exceptions and logs pipeline runs."""
 
         def wrapped_handler() -> None:
             start_time = time.time()
             started_at = datetime.datetime.now(datetime.UTC)
 
+            idempotency_key_value = None
+
+            # Check if an idempotency key function is given, and skip this run if there was a
+            # previous successful run of the pipeline with the same key.
+            if idempotency_key:
+                idempotency_key_value = idempotency_key()
+                count = (
+                    select(func.count())
+                    .select_from(PipelineRun)
+                    .where(PipelineRun.idempotency_key == idempotency_key_value)
```

Comment:
As you said in the comment above yourself, I think it would be nicer to make the naming more explicit. Especially for the `Callable`, something like `idempotency_key_function` would already help a lot I think, does not need to be sophisticated. Otherwise, it becomes really confusing, especially with something like line 195. Here, we could also think about renaming the property on the `PipelineRun`, but that's of lower importance IMO.